docs/docs/query-filters.md

```diff
@@ -0,0 +1,171 @@
+- `ne`
+  - This is like `filterValue !== resultValue` in JavaScript
+  - Can only be used with numbers, strings, booleans, and null
+  - Strict comparison except for `null`
+  - Will always return any nodes with partial paths
+- `in`
+  - This is like `filterValue.includes(resultValue)` in JavaScript
+  - Can only be used with an array of numbers, strings, booleans, and null
+  - Strict comparison
+  - When matching `null`, and only then, it also returns all nodes with partial paths
+- `nin`
+  - This is like `!filterValue.includes(resultValue)` in JavaScript
+  - Can only be used with an array of numbers, strings, booleans, and null
+  - Strict comparison
+  - Will always return any nodes with partial paths
+- `lt`, `lte`, `gt`, `gte`
+  - This is like `<`, `<=`, `>`, `>=` respectively
+  - Can only be used with numbers, strings, booleans, and null
+  - If filter value is `null`;
+    - `lt` and `gt` will never match anything
+    - `lte` and `gte` will only match nodes with a result value of exactly `null`
+  - Will never return nodes with partial paths
+  - Weak comparison due to the nature of the JavaScript operators
+- `regex`, `glob`
+  - This is like `new RegExp(filterValue).test(resultValue)` (with caveats for the filterValue syntax)
+  - Glob pattern is converted to a JavaScript RegExp with [micromatch](https://github.com/micromatch/micromatch)
+  - The `regex` filterValue must be a stringified regular expression, including leading and trailing forward slash and optional flags; Like `"/foo/g"`
+  - Never returns nodes with partial paths
+  - While testing, result values are explicitly casted to a string through `String(resultValue)` before passing it to `regex.test()`
+
+### Nulls and partial paths
+
+Due to legacy support for MongoDB compatibility, there are edge cases for each comparator when it comes to `null` values and for partial or non-existing paths. It's best to try and avoid these cases altogether.
+
+## Performance
+
+The key metric that impacts the performance of your queries is the node count for your type of nodes. You can see a dump of these counts when using the `--verbose` flag while building (`gatsby build --verbose`). It tells you the node counts per type during bootstrap. There will be a separate message of the number of page nodes, since they are generated later.
+
+While the number of pages is definitely a factor for the number of nodes, there can be other factors at play that cause many internal nodes even when the actual number of pages seems to be low.
+
+The actual performance of filtering is a combination of the number of nodes you have to search through, the kind of comparator being used, and how many unique filters a query consists of.
+
+For a low node count none of this matters. Roughly speaking, a site with fewer than 1000 - 10000 nodes should not have to worry too much about this.
+
+When scaling up these are some rules of thumb to keep in mind when filtering:
```

Comment:
"Rules of thumb" is one of those phrases we should try to avoid if possible, open to other suggestions.

```suggestion
When scaling up these are some rules to keep in mind when filtering:
```

Reply:
Ok